check_api/src/main/java/com/google/errorprone/ErrorPronePlugins.java

```diff
@@ -45,11 +54,50 @@ public static ScannerSupplier loadPlugins(ScannerSupplier scannerSupplier, Conte
       return scannerSupplier;
     }
     ClassLoader loader = fileManager.getClassLoader(StandardLocation.ANNOTATION_PROCESSOR_PATH);
-    Iterable<BugChecker> extraBugCheckers = ServiceLoader.load(BugChecker.class, loader);
-    if (Iterables.isEmpty(extraBugCheckers)) {
+
+    List<Class<? extends BugChecker>> bugCheckerClasses;
+    // Synchronize to prevent races
+    synchronized(ErrorPronePlugins.class) {
+      // disable URL caching to avoid stale state bugs when run within a daemon-building process
+      boolean oldURLCaching = setUrlCaching(false);
+      Iterable<BugChecker> extraBugCheckers = ServiceLoader.load(BugChecker.class, loader);
+      bugCheckerClasses = Lists.newArrayList(Iterables.transform(extraBugCheckers, GET_CLASS));
+      // restore the old URL caching setting
+      setUrlCaching(oldURLCaching);
+    }
+    if (bugCheckerClasses.isEmpty()) {
       return scannerSupplier;
+    } else {
+      return scannerSupplier.plus(ScannerSupplier.fromBugCheckerClasses(bugCheckerClasses));
+    }
+  }
+
+  /**
+   * A dummy {@link URLConnection} object used for disabling the global setting to use caches
+   */
+  private static URLConnection DUMMY_CONNECTION = null;
+
+  static {
+    try {
+      DUMMY_CONNECTION = new URLConnection(new URL("file:///")) {
+        @Override
+        public void connect() throws IOException {
+
+        }
+      };
+    } catch (MalformedURLException e) {
+      // this should never fail
+      throw new RuntimeException(e);
```

Comment:
I'm a bit leery about throwing a runtime exception in the class init block (although I understand that it 'should be impossible')

Could you try constructing this on-demand in your set/getUrlCaching methods, and replace RuntimeException with AssertionError (to point out its impossibility)?